backup/backup.go

```diff
@@ -0,0 +1,123 @@
+package backup
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	help "github.com/featureform/helpers"
+	"go.etcd.io/etcd/api/v3/mvccpb"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"io/ioutil"
+	"time"
+)
+
+type ProviderType string
+
+const (
+	AZURE ProviderType = "AZURE"
+	LOCAL              = "LOCAL"
+)
+
+// Client Allows ETCD to be tested with mock values
+type Client interface {
+	Get(ctx context.Context, key string, opts ...clientv3.OpOption) (*clientv3.GetResponse, error)
+}
+
+type Backup struct {
+	ETCDClient   Client
+	ProviderType ProviderType
+}
+
+func (b *Backup) Save(name string) error {
+	err := b.takeSnapshot(name)
+	if err != nil {
+		return fmt.Errorf("could not take snapshot: %v", err)
+	}
+
+	backupProvider, err := b.getBackupProvider()
+	if err != nil {
+		return err
+	}
+	err = backupProvider.Init()
+	if err != nil {
+		return fmt.Errorf("could not initialize provider: %v", err)
+	}
+	err = backupProvider.Upload(name, name)
+	if err != nil {
+		return fmt.Errorf("cannot upload snapshot to filestore: %v", err)
+	}
+	return nil
+}
+
+func (b *Backup) Restore() error {
+	return nil
```

Comment:
Not implemented yet on purpose yeah? Random note but I like to throw in a panic instead of a return nil if I don't implement something to avoid forgetting. But I am very forgetful